lib/models/LifecycleConfiguration.ts

```diff
@@ -703,14 +706,41 @@ export default class LifecycleConfiguration {
                 'NoncurrentDays');
             return { error };
         }
+
+        const actionParams: {
+            error?: ArsenalError;
+            days: number;
+            newerNoncurrentVersions: number;
+        } = {
+            days: 0,
+            newerNoncurrentVersions: 0,
+        };
+
         const daysInt = parseInt(subNVExp.NoncurrentDays[0], 10);
         if (daysInt < 1) {
             const msg = 'NoncurrentDays is not a positive integer';
             const error = errors.InvalidArgument.customizeDescription(msg);
             return { error };
         } else {
-            return { days: daysInt };
+            actionParams.days = daysInt;
+        }
+
+        if (subNVExp.NewerNoncurrentVersions) {
```

Comment:
shouldn't we remove the check at line 703 ` if (!subNVExp.NoncurrentDays)`, now that we introduced a new type of check, or make sure at least one of the two checks is present instead? According to https://docs.aws.amazon.com/AmazonS3/latest/API/API_NoncurrentVersionExpiration.html, it is not required to have `NoncurrentDays` attribute.

Reply:
I have completed  missed this case and that `NoncurrentDays` and `NewerNoncurrentVersions` are independent from the other.